scripts/cmake/utilities.cmake

```diff
@@ -19,3 +22,437 @@ function(mz_target_handle_warnings MZ_TARGET)
+
+    list(APPEND LOCAL_LINK_LIBRARIES ${MZ_GENERATE_LINK_LIBRARIES_EXTRA_DEPENDENCIES})
+
+    # 6. Test dependency handling
+
+    set(LOCAL_TEST_LINK_LIBRARIES ${LOCAL_LINK_LIBRARIES})
+
+    # Replace dependencies for test dependencies that start with `replace-`
+    set(REPLACER_DEPENDENCIES ${MZ_GENERATE_LINK_LIBRARIES_TEST_DEPENDENCIES})
+    list(FILTER REPLACER_DEPENDENCIES INCLUDE REGEX "^\\r\\e\\p\\l\\a\\c\\e\\-")
+    foreach(REPLACER_DEPENDENCY ${REPLACER_DEPENDENCIES})
+        # Get the name of the original dependency
+        string(REPLACE "replace-" "" ORIGINAL_DEPENDENCY ${REPLACER_DEPENDENCY})
+        # Remove it  from the list
+        list(REMOVE_ITEM LOCAL_TEST_LINK_LIBRARIES ${ORIGINAL_DEPENDENCY})
+    endforeach()
+
+    # Tests always need Qt6::Test
+    find_package(Qt6 REQUIRED COMPONENTS Test)
+    list(APPEND LOCAL_TEST_LINK_LIBRARIES Qt6::Test)
+
+    # Finalize the list of test dependencies.
+    list(APPEND LOCAL_TEST_LINK_LIBRARIES ${MZ_GENERATE_LINK_LIBRARIES_TEST_DEPENDENCIES})
+
+    # 7. Set the lists in the parent scope.
+
+    set(LINK_LIBRARIES ${LOCAL_LINK_LIBRARIES} PARENT_SCOPE)
+    set(TEST_LINK_LIBRARIES ${LOCAL_TEST_LINK_LIBRARIES} PARENT_SCOPE)
+endfunction()
+
+# Sets the following new variables in the parent scope:
+#
+# ALL_SOURCES
+# APPLE_SOURCES -- These are no included in ALL_SOURCES.
+function(mz_generate_sources_list)
+    cmake_parse_arguments(
+        MZ_GENERATE_SOURCES_LIST # prefix
+        "" # options
+        "" # single-value args
+        "SOURCES;IOS_SOURCES;ANDROID_SOURCES;MACOS_SOURCES;LINUX_SOURCES;WINDOWS_SOURCES;WASM_SOURCES;DUMMY_SOURCES" # multi-value args
+        ${ARGN})
+
+    set(LOCAL_ALL_SOURCES ${MZ_GENERATE_SOURCES_LIST_SOURCES})
+
+    # 1. Handle platform specific sources
+
+    if(${MZ_PLATFORM_NAME} STREQUAL "ios")
+        list(APPEND LOCAL_ALL_SOURCES ${MZ_GENERATE_SOURCES_LIST_IOS_SOURCES})
+    elseif(${MZ_PLATFORM_NAME} STREQUAL "android")
+        list(APPEND LOCAL_ALL_SOURCES ${MZ_GENERATE_SOURCES_LIST_ANDROID_SOURCES})
+    elseif(${MZ_PLATFORM_NAME} STREQUAL "macos")
+        list(APPEND LOCAL_ALL_SOURCES ${MZ_GENERATE_SOURCES_LIST_MACOS_SOURCES})
+    elseif(${MZ_PLATFORM_NAME} STREQUAL "linux")
+        list(APPEND LOCAL_ALL_SOURCES ${MZ_GENERATE_SOURCES_LIST_LINUX_SOURCES})
+    elseif(${MZ_PLATFORM_NAME} STREQUAL "windows")
+        list(APPEND LOCAL_ALL_SOURCES ${MZ_GENERATE_SOURCES_LIST_WINDOWS_SOURCES})
+    elseif(${MZ_PLATFORM_NAME} STREQUAL "wasm")
+        list(APPEND LOCAL_ALL_SOURCES ${MZ_GENERATE_SOURCES_LIST_WASM_SOURCES})
+    elseif(${MZ_PLATFORM_NAME} STREQUAL "dummy")
+        list(APPEND LOCAL_ALL_SOURCES ${MZ_GENERATE_SOURCES_LIST_DUMMY_SOURCES})
+    else()
+        message(FATAL_ERROR "MZ_PLATFORM_NAME must be set before creating modules.")
+    endif()
+
+    # 2. Separate out Swift and ObjC sources
+
+    set(SWIFT_SOURCES ${LOCAL_ALL_SOURCES})
+    list(FILTER SWIFT_SOURCES INCLUDE REGEX "(.*)\\s\\w\\i\\f\\t$")
```

Comment:
I am not sure about this regex. If the goal is to separate `.swift` and `.mm` files, shouldn't this be:
`list(FILTER SWIFT_SOURCES INCLUDE REGEX "\\.swift$")`

The dot is a special character, so `(.*)` matches anything (including stuff without a literal dot). As is, I think this would match a string like `example.foobarswift`

Reply:
Yeah, honestly no idea why I am escaping each character here. Good call on the `.` as well :facepalm: